test/functional/test_esxi.py

```diff
@@ -0,0 +1,195 @@
+import unittest
+import os
+import time
+import subprocess
+import yaml
+from test import fixtures
+import re
+from infrasim import model
+from infrasim import InfraSimError
+import paramiko
+from infrasim.package_install import install_bintray_packages
+import urllib
+
+old_path = os.environ.get("PATH")
+new_path = "{}/bin:{}".format(os.environ.get("PYTHONPATH"), old_path)
+source_esxi_path = "ftp://10.62.59.23/idic/iso/esxi_img/esxi6p3-1.qcow2"
+esxi_path = "/tmp/esxi6p3-1.qcow2"
+tmp_conf_file = "/tmp/test.yml"
+conf = {}
+
+
+def run_command(cmd="", shell=True, stdout=None, stderr=None, interactive_input=""):
+    child = subprocess.Popen(cmd, shell=shell,
+                             stdout=stdout, stderr=stderr)
+    cmd_result = child.communicate(interactive_input)
+    cmd_return_code = child.returncode
+    if cmd_return_code != 0:
+        return -1, cmd_result[1]
+    return 0, cmd_result[0]
+
+def read_buffer(channel):
+    while not channel.recv_ready():
+        continue
+    str_output = ''
+    str_read = ''
+    while True:
+        str_read = str(channel.recv(40960))
+        str_output += str_read
+        prompts = re.findall(r'root@localhost:~', str_output)
+        if len(prompts) == 2:
+            break
+        time.sleep(1)
+    return str_output
+
+def start_node():
+    global conf
+    fake_config = fixtures.FakeConfig()
+    conf = fake_config.get_node_info()
+
+    conf["compute"]["storage_backend"] = [
+        {"type": "ahci",
+         "max_drive_per_controller": 6,
+         "drives": [{"bootindex":1, "size": 8, "file": esxi_path}]},
+        {"type": "lsisas3008",
+         "max_drive_per_controller": 6,
+         "drives": [{"size": 8, "file": "/tmp/sda.img"}]}
+    ]
+    conf["compute"]["memory"]["size"] = 4096
+    conf["compute"]["cpu"]["features"] = "+vmx"
+    conf["compute"]["networks"][0]["device"] = "e1000"
+    conf["compute"]["networks"][0]["network_mode"] = "nat"
+
+    with open(tmp_conf_file, "w") as yaml_file:
+        yaml.dump(conf, yaml_file, default_flow_style=False)
+
+    result = run_command("qemu-img", True, subprocess.PIPE, subprocess.PIPE)[1]
+    if "qemu-img: error while loading shared libraries: libaio.so.1" in result:
```

Comment:
The test costs 2 minutes including downloading. But it might be slower for devs outside our company. 
Bintray may not be a chioce since it's 800+M.
github large file is a good idea, I can have a look at it.